templates/tanstack-shell/app/components/RegisterForm.tsx

```
@@ -0,0 +1,226 @@
+    onSubmit: async ({ value }) => {
+      try {
+        await register({ data: value })
+        router.invalidate()
+        if (onSuccess) {
+          onSuccess()
+        } else {
+          await router.navigate({ to: '/dashboard' })
+        }
+      } catch (error) {
+        // Form will handle the error display
+        throw error
```

Biome (1.9.4)

[error] The catch clause that only rethrows the original error is useless.

An unnecessary catch clause can be confusing.
Unsafe fix: Remove the try/catch clause.


(lint/complexity/noUselessCatch)